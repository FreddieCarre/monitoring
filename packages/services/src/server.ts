import { ApolloServer } from 'apollo-server';
import { logger } from '../utils/logger';

import {
  typeDefs as EnergyTypes,
  resolvers as EnergyResolvers
} from './graphql/energyConsumption';
import {
  typeDefs as WeatherTypes,
  resolvers as WeatherResolvers
} from './graphql/weather';
import { Query } from './graphql';

export const server = new ApolloServer({
  typeDefs: [Query, EnergyTypes, WeatherTypes],
  resolvers: [EnergyResolvers, WeatherResolvers]
});

server.listen()
  .then(({ url }) => {
    logger.info({ url }, 'Server is listening');
  })
  .catch((error: Error) => logger.error({
    msg: 'Server caught error',
    error: error.message,
    stack: error.stack
  }));
