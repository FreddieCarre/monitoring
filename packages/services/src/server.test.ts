import { gql } from 'apollo-server';
import { createTestClient } from 'apollo-server-testing';

import { server } from './server';

const ENERGY_CONSUMPTION = gql`
  query ExampleQuery {
    energyConsumption {
      Timestamp
      Consumption
    }
  }
`;
describe('Server', () => {
  it('should return an empty array when no points found', async () => {
    const { query } = createTestClient(server);

    const res = await query({
      query: ENERGY_CONSUMPTION,
      variables: {
        start: '2020-04-01T00:00:00.000Z',
        stop: '2020-04-01T23:59:59.000Z'
      }
    });
    expect(res).toMatchSnapshot();
  });
});
