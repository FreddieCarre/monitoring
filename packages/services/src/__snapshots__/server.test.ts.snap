// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Server should return an empty array when no points found 1`] = `
Object {
  "data": Object {
    "energyConsumption": null,
  },
  "errors": Array [
    [GraphQLError: Invalid time value],
  ],
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;

exports[`Server should return an empty array when no points within 2020-01-03 1`] = `
Object {
  "data": Object {
    "energyConsumption": Array [
      Object {
        "Consumption": 4.43,
        "Timestamp": "2020-01-03T00:00:00Z",
      },
      Object {
        "Consumption": 5.31,
        "Timestamp": "2020-01-03T00:30:00Z",
      },
      Object {
        "Consumption": 5.56,
        "Timestamp": "2020-01-03T01:00:00Z",
      },
      Object {
        "Consumption": 4.81,
        "Timestamp": "2020-01-03T01:30:00Z",
      },
      Object {
        "Consumption": 0.66,
        "Timestamp": "2020-01-03T02:00:00Z",
      },
      Object {
        "Consumption": 13.02,
        "Timestamp": "2020-01-03T02:30:00Z",
      },
      Object {
        "Consumption": 5.37,
        "Timestamp": "2020-01-03T03:00:00Z",
      },
      Object {
        "Consumption": 2.32,
        "Timestamp": "2020-01-03T03:30:00Z",
      },
      Object {
        "Consumption": 5.29,
        "Timestamp": "2020-01-03T04:00:00Z",
      },
      Object {
        "Consumption": 11.94,
        "Timestamp": "2020-01-03T04:30:00Z",
      },
      Object {
        "Consumption": 9.18,
        "Timestamp": "2020-01-03T05:00:00Z",
      },
      Object {
        "Consumption": 8.09,
        "Timestamp": "2020-01-03T05:30:00Z",
      },
      Object {
        "Consumption": 11.8,
        "Timestamp": "2020-01-03T06:00:00Z",
      },
      Object {
        "Consumption": 3.48,
        "Timestamp": "2020-01-03T06:30:00Z",
      },
      Object {
        "Consumption": 6.52,
        "Timestamp": "2020-01-03T07:00:00Z",
      },
      Object {
        "Consumption": 3.5,
        "Timestamp": "2020-01-03T07:30:00Z",
      },
      Object {
        "Consumption": 10.6,
        "Timestamp": "2020-01-03T08:00:00Z",
      },
      Object {
        "Consumption": 5.38,
        "Timestamp": "2020-01-03T08:30:00Z",
      },
      Object {
        "Consumption": 19.5,
        "Timestamp": "2020-01-03T09:00:00Z",
      },
      Object {
        "Consumption": 29,
        "Timestamp": "2020-01-03T09:30:00Z",
      },
      Object {
        "Consumption": 30.76,
        "Timestamp": "2020-01-03T10:00:00Z",
      },
      Object {
        "Consumption": 8.04,
        "Timestamp": "2020-01-03T10:30:00Z",
      },
      Object {
        "Consumption": 15.44,
        "Timestamp": "2020-01-03T11:00:00Z",
      },
      Object {
        "Consumption": 7.46,
        "Timestamp": "2020-01-03T11:30:00Z",
      },
      Object {
        "Consumption": 2.1,
        "Timestamp": "2020-01-03T12:00:00Z",
      },
      Object {
        "Consumption": 7.13,
        "Timestamp": "2020-01-03T12:30:00Z",
      },
      Object {
        "Consumption": 7.29,
        "Timestamp": "2020-01-03T13:00:00Z",
      },
      Object {
        "Consumption": 5.55,
        "Timestamp": "2020-01-03T13:30:00Z",
      },
      Object {
        "Consumption": 0.24,
        "Timestamp": "2020-01-03T14:00:00Z",
      },
      Object {
        "Consumption": 8.68,
        "Timestamp": "2020-01-03T14:30:00Z",
      },
      Object {
        "Consumption": 1.25,
        "Timestamp": "2020-01-03T15:00:00Z",
      },
      Object {
        "Consumption": 7.26,
        "Timestamp": "2020-01-03T15:30:00Z",
      },
      Object {
        "Consumption": 1.21,
        "Timestamp": "2020-01-03T16:00:00Z",
      },
      Object {
        "Consumption": 2.02,
        "Timestamp": "2020-01-03T16:30:00Z",
      },
      Object {
        "Consumption": 1.7,
        "Timestamp": "2020-01-03T17:00:00Z",
      },
      Object {
        "Consumption": 13.96,
        "Timestamp": "2020-01-03T17:30:00Z",
      },
      Object {
        "Consumption": 8.44,
        "Timestamp": "2020-01-03T18:00:00Z",
      },
      Object {
        "Consumption": 8.45,
        "Timestamp": "2020-01-03T18:30:00Z",
      },
      Object {
        "Consumption": 1.4,
        "Timestamp": "2020-01-03T19:00:00Z",
      },
      Object {
        "Consumption": 14.76,
        "Timestamp": "2020-01-03T19:30:00Z",
      },
      Object {
        "Consumption": 15.39,
        "Timestamp": "2020-01-03T20:00:00Z",
      },
      Object {
        "Consumption": 7.08,
        "Timestamp": "2020-01-03T20:30:00Z",
      },
      Object {
        "Consumption": 9.78,
        "Timestamp": "2020-01-03T21:00:00Z",
      },
      Object {
        "Consumption": 12.48,
        "Timestamp": "2020-01-03T21:30:00Z",
      },
      Object {
        "Consumption": 9.67,
        "Timestamp": "2020-01-03T22:00:00Z",
      },
      Object {
        "Consumption": 2.95,
        "Timestamp": "2020-01-03T22:30:00Z",
      },
      Object {
        "Consumption": 10.6,
        "Timestamp": "2020-01-03T23:00:00Z",
      },
      Object {
        "Consumption": 6.81,
        "Timestamp": "2020-01-03T23:30:00Z",
      },
    ],
  },
  "errors": undefined,
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;

exports[`Server should return an empty array when no points within 2020-01-04 1`] = `
Object {
  "data": Object {
    "energyConsumption": null,
  },
  "errors": Array [
    [GraphQLError: Invalid time value],
  ],
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;
